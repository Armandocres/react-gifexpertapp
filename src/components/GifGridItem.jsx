import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ title, url }) => {
	return (
		<div className="card animate__bounceIn">
			<figure className="card__image">
				<img src={url} alt={title} />
			</figure>
			<p>{title}</p>
		</div>
	);
};

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};

export default GifGridItem;
