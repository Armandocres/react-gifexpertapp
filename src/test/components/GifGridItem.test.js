import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en el componente GifGrid', () => {
  const title = 'un titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem  title={title} url={url} />);

  test('Se renderiza componente?', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de tener un parrafo con el titulo', () => {
    const p = wrapper.find('p');
    expect( p.text().trim()).toBe(title);
  });

  test('Debe de obtener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');

    expect( img.prop('src')).toBe( url );
    expect( img.prop('alt')).toBe( title );

  })

  test('Debe tener animate__bounceIn', () => {
    const div = wrapper.find('div');

    expect(div.hasClass('animate__bounceIn')).toBe( true );
  });
});
