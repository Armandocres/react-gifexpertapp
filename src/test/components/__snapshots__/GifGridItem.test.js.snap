// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Se renderiza componente? 1`] = `
<div
  className="card animate__bounceIn"
>
  <figure
    className="card__image"
  >
    <img
      alt="un titulo"
      src="https://localhost/algo.jpg"
    />
  </figure>
  <p>
    un titulo
  </p>
</div>
`;
